simple-weather/simple-weather/Model/CurrentWeather.swift

```diff
@@ -6,13 +6,47 @@
 //  Copyright © 2019 egmars.janis.timma. All rights reserved.
 //
 
-//import Foundation
-//
-//class CurrentWeather {
-//    
-//    private var _cityName: String!
-//    private var _date: String!
-//    private var _weatherType: String!
-//    private var _currentTemp: Double!
-//    
-//}
+import Foundation
+
+class CurrentWeather {
+    
+    private var _cityName: String!
+    private var _date: String!
+    private var _weatherType: String!
+    private var _currentTemp: Double!
+    
+    
+    var cityName: String {
```

Comment:
I really doubt this is approach was your decision.
Remember to prioritise understanding over completing.

**Explain:**
Why use this approach
```swift
private var _cityName: String!
var cityName: String {
    if _cityName == nil {
        _cityName = ""
    }
    return _cityName
}
```
instead of this
```swift
var cityName: String = ""
```

Reply:
Of course it wasn't. I found this approach in tutorial where this code was described as good example to keep dates private in class and later convert them to be accessible to another classes with check if contains nil.

Reply:
@vm1000 Good, so it prevents developers to accidentally set value outside of particular class and, secondly, it safely unwraps optional values and returns default value if value was nil.

However, we could simply use `private(set) var` which means value to this variable can be set only in current scope (in our case only within `CurrentWeather` class).
Secondly, in this particular case, almost all values will be used to set given `title` to `label.text`, and text has this check already, it accepts `String?` value and if it is `nil` removes `label` (same as `label.text = ""` or `label.text = nil`). This approach is useful where values should be manipulated before returning those to `private var`, in our case API will return degrees in fahrenheit, which we will have to convert to celsius therefore computed getter could be useful in our case

Reply:
So in current case there is no need to check values for nil in this class and no need to convert values to public var?

Reply:
In our case it will not be an issue if values are un-nililable. Usually nils are unwrapped when they need to be used, for example, there could be a case where we want to check
```swift
if let name = cityName {
    // do something with name
}
```
In your approach that would not be possible